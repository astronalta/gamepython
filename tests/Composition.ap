# aptest exec on

TestComp1 < Object {
    var1 Int = 42
    share Int 

    test1() String {
        ret "hello\n"
    }
}

TestComp2 < Object {
    var2 Int = 99
    share Int

    test2() String {
        ret "goodbye\n"
    }
}

TestClass < Object {
    TestComp1
    TestComp2
}

TestInterface < Interface {
    test1() String
    test2() String
}

main() Int {
    Io::print(TestClass().test1())
    Io::print(TestClass().test2())
    Io::println(TestClass().var1.str)
    Io::println(TestClass().var2.str)

    obj TestInterface = TestClass()
    Io::print(obj.test1())
    Io::print(obj.test2())
    Io::stdout().flush()
    TestClass().hash.str
    ret 0 
}
