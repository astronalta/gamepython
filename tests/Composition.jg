# jgtest exec on

TestComp1 < Object {
    var1 Int = 42
    share Int 

    @init() {
        Io::println("TestComp1()")
    }

    test1() String {
        ret "hello\n"
    }
}

TestComp2 < Object {
    var2 Int = 99
    share Int

    test2() String {
        ret "goodbye\n"
    }
}

TestClass < Object {
    TestComp1
    TestComp2
}

TestInterface < Interface {
    test1() String
    test2() String
}

main() Int {
    obj = TestClass()
    Io::print(obj.test1())
    Io::print(obj.test2())
    Io::println(obj.var1.str)
    Io::println(obj.var2.str)
    obj.hash.str

    intf TestInterface = TestClass()
    Io::print(intf.test1())
    Io::print(intf.test2())
    Io::stdout().flush()
    ret 0 
}
