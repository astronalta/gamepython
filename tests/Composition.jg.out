TestComp1()
hello
goodbye
42
99
TestComp1()
hello
goodbye
